src/core/border_router/routing_manager.cpp

```diff
@@ -221,14 +240,70 @@ Error RoutingManager::LoadOrGenerateRandomBrUlaPrefix(void)
 }
 
 #if OPENTHREAD_CONFIG_BORDER_ROUTING_NAT64_ENABLE
+const otIp4Address RoutingManager::kWellKnownIpv4OnlyAddress1 = {{{192, 0, 0, 170}}};
+const otIp4Address RoutingManager::kWellKnownIpv4OnlyAddress2 = {{{192, 0, 0, 171}}};
+
+void RoutingManager::DiscoverInfraIfNat64Prefix(void)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(IsInitialized() && mInfraIf.IsRunning(), error = kErrorInvalidState);
+
+    LogInfo("Discovering infrastructure NAT64 prefix on %s", mInfraIf.ToString().AsCString());
+    error = mInfraIf.DiscoverNat64Prefixes();
+
+exit:
+    if (error != kErrorNone)
+    {
+        LogWarn("Failed to request infrastructure NAT64 prefix on %s: %s", mInfraIf.ToString().AsCString(),
+                ErrorToString(error));
+    }
+}
+
+void RoutingManager::DiscoverInfraIfNat64PrefixesDone(const Ip6::Address *aAddresses, uint8_t aNumAddresses)
+{
+    Ip6::Prefix   prefix;
+    const uint8_t prefixLength[] = {32, 40, 48, 56, 64, 96};
+
+    prefix.Clear();
+    for (uint8_t i = 0; aAddresses != nullptr && i < aNumAddresses && !prefix.IsValidNat64(); i++)
+    {
+        for (uint8_t length : prefixLength)
+        {
+            Ip4::Address ip4Address;
+
+            ip4Address.ExtractFromIp6Address(length, aAddresses[i]);
+            if (memcmp(ip4Address.GetBytes(), kWellKnownIpv4OnlyAddress1.mFields.m8, OT_IP4_ADDRESS_SIZE) == 0 ||
+                memcmp(ip4Address.GetBytes(), kWellKnownIpv4OnlyAddress2.mFields.m8, OT_IP4_ADDRESS_SIZE) == 0)
```

Comment:
Can use `==` to compare `Ip4::Address`. 

`Ip4::Address` inherits from `Equatable` which adds the overload of `operator==`.
May need to use `AsCoreType()` on `kWellKnownIpv4OnlyAddress1` (since they are C struct type`otIp4Address` and not `Ip4::Address`).


```suggestion
            if ((ip4Address == AsCoreType(&kWellKnownIpv4OnlyAddress1) || (ip4Address == AsCoreType(&kWellKnownIpv4OnlyAddress2))

Reply:
Updated.